typescript/sdk/src/deploy/middleware/LiquidityLayerApp.ts

```diff
@@ -38,68 +37,66 @@ const PortalBridgedTokenTopic = PortalAdapterInterface.getEventTopic(
   PortalAdapterInterface.getEvent('BridgedToken'),
 );
 
-interface CircleBridgeMessage<Chain> {
-  chain: Chain;
-  remoteChain: Chain;
+interface CircleBridgeMessage {
+  chain: ChainName;
+  remoteChain: ChainName;
   txHash: string;
   message: string;
   nonce: number;
   domain: number;
   nonceHash: string;
 }
 
-interface PortalBridgeMessage<Chain> {
-  origin: Chain;
+interface PortalBridgeMessage {
+  origin: ChainName;
   nonce: number;
   portalSequence: number;
-  destination: Chain;
+  destination: ChainName;
 }
 
-export class LiquidityLayerApp<
-  Chain extends ChainName = ChainName,
-> extends HyperlaneApp<LiquidityLayerContracts, Chain> {
+export class LiquidityLayerApp extends HyperlaneApp<LiquidityLayerContracts> {
   constructor(
-    public readonly contractsMap: ChainMap<Chain, LiquidityLayerContracts>,
-    public readonly multiProvider: MultiProvider<Chain>,
-    public readonly config: ChainMap<Chain, BridgeAdapterConfig>,
+    public readonly contractsMap: ChainMap<LiquidityLayerContracts>,
+    public readonly multiProvider: MultiProvider,
+    public readonly config: ChainMap<BridgeAdapterConfig>,
   ) {
     super(contractsMap, multiProvider);
   }
 
-  async fetchCircleMessageTransactions(chain: Chain): Promise<string[]> {
-    const cc = this.multiProvider.getChainConnection(chain);
+  async fetchCircleMessageTransactions(chain: ChainName): Promise<string[]> {
     const params = new URLSearchParams({
       module: 'logs',
       action: 'getLogs',
       address: this.getContracts(chain).circleBridgeAdapter!.address,
       topic0: BridgedTokenTopic,
     });
-    const req = await fetch(`${cc.getApiUrl()}?${params}`);
+    const url = `${this.multiProvider.getExplorerApiUrl(chain)}?${params}`;
```

Comment:
That would bake the query engine (e.g. fetch) assumption directly into the MP, which potentially causes problems for using it older versions of Node (which are still the most commonly used versions)

Reply:
I'm open to refactor existing uses of these methods to break the SDK's core code's dependency on them

Reply:
Sorry I should have been more clear, I'm not suggesting to move the fetching into the MP, instead I'm suggesting to move the construction of the request to MP. Here for example, we are assuming that there is an explorer url set, but what if it is not? IMO, it should be a `getExplorerRequest(chain, params)` that either returns null or the request that can then be made?

While we are here, it does remind me that we should include the API key in the chain metadata/explorer object so that we can get ridd of the special casing for `ContractVerifier`

My point about `ticket` was that I have a bunch of suggestions during this review which don't have to happen within the scope of this PR, I just wanted to make sure that we don't forget about them. If you don't end up doing this, would you mind letting me know and I'll file a ticket?

Reply:
Punting: https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/1831